onelogin_rules/onelogin_unusual_login.py

```diff
@@ -0,0 +1,60 @@
+import json
+import requests
+from panther_oss_helpers import get_string_set, put_string_set
+
+
+def rule(event):
+    # Pre-filter to save compute time where possible. event_type_id = 5 is login events.
+    if event.get('event_type_id') != 5 or event.get('ipaddr') is None:
+        return False
+
+    # Lookup geo-ip data via API call
+    url = 'https://ipinfo.io/' + event['ipaddr'] + '/geo'
+
+    # Skip API call if this is a unit test
+    if __name__ == 'PolicyApiTestingPolicy':
+        resp = lambda: None
+        setattr(resp, 'status_code', 200)
+        setattr(resp, 'text', event['api_data'])
+    else:
+        resp = requests.get(url)
+
+    if resp.status_code != 200:
+        # Could raise an exception here for ops team to look into
+        return False
+    login_info = json.loads(resp.text)
+    login_tuple = login_info.get('region', '<REGION>') + ":" + login_info.get(
+        'city', '<CITY>')
+
+    # Lookup & store persistent data
+    event_key = get_key(event)
+    last_login_info = get_string_set(event_key)
+    if not last_login_info:
+        # Store this as the first login if we've never seen this user login before
+        put_string_set(event_key, [json.dumps({login_tuple: 1})])
+        return False
+    last_login_info = json.loads(last_login_info.pop())
```

Comment:
I'm having a hard time picturing what this is. A set of region strings? Perhaps a comment with an example response from the API would be helpful

Reply:
Ah good point, a lot of the context was initially discussed verbally while working on this rule. I'll add some comments to transfer that context into code.

Reply:
I added a comment about what the rule is trying to accomplish.